src/error.rs

```diff
@@ -7,6 +7,8 @@ pub enum ODataError {
     #[error(transparent)]
     UnsupportedDataType(#[from] UnsupportedDataType),
     #[error(transparent)]
+    UnsupportedColumnType(#[from] UnsupportedColumnType),
```

Comment:
Not sure how this error variant is different from `UnsupportedDataType` - it carries the same information. Does it add any value or can we simply return `UnsupportedDataType` in that function?

Reply:
I think we need to use a different type from `UnsupportedDataType`, that would be helpful for debugging, but yeah not sure about the name I chose `UnsupportedColumnType` though. Maybe we could choose a different name for both of them to clarify where the error is coming from.

Reply:
I think solving the problem of _"where the error is coming from"_ by giving errors different names is a messy approach. To me this sounds almost like creating error types like `UnsupportedDataType_FileA_Line123`.

A better solution would be to include a `Backtrace` into errors - that's what we do everywhere in our code, but this requires unfortunately still requires `nightly`.